Core/Core/Inbox/ViewModel/InboxViewModel.swift

```diff
@@ -135,11 +136,16 @@ public class InboxViewModel: ObservableObject {
             .store(in: &subscriptions)
     }
 
-    private func subscribeToMenuTapEvents(router: Router) {
+    private func subscribeToTapEvents(router: Router) {
         menuDidTap
             .sink { [weak router] source in
                 router?.route(to: "/profile", from: source, options: .modal())
             }
             .store(in: &subscriptions)
+        messageDidTap
+            .sink { [weak router] (messageID, controller) in
```

Comment:
I think we can capture `router` strongly here (and 5 lines above :)). It's a shared instance so it won't ever deallocate and it's a local variable so it won't create a retain cycle.